src/sidebar/components/ShareDialog/ImportAnnotations.tsx

```diff
@@ -159,20 +168,27 @@ function ImportAnnotations({
           </Select>
         </>
       )}
-      {busy && <LoadingSpinner />}
       {error && (
         // TODO - Add a support link here.
         <p data-testid="error-info">
           <b>Unable to find annotations to import:</b> {error}
         </p>
       )}
       <CardActions>
+        {importProgress !== null && (
+          <span data-testid="progress-text" className="text-grey-6">
+            {importProgress}% complete
+          </span>
+        )}
         <Button
           data-testid="import-button"
           disabled={!importReady || busy}
           onClick={importAnnotations}
           variant="primary"
         >
+          {importProgress !== null && (
+            <CircularProgress size={22} value={importProgress} />
+          )}
```

Comment:
The button size increases slightly during an import. Initially I thought this might look jarring, so we'd need to pad the button when no import is active. In practice I found it looked better to me if we didn't pad in the idle state, and just let the small size change happen.